src/media/js/addon/actions/addon.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+import {createAction} from 'redux-actions';
+import req from 'request';
+import Url from 'urlgray';
+import urlJoin from 'url-join';
+
+
+export const FETCH_OK = 'ADDON_ADDON__FETCH_OK';
+const fetchOk = createAction(FETCH_OK);
+
+export const FETCH_VERSIONS_OK = 'ADDON_ADDON__FETCH_VERSIONS_OK';
+const fetchVersionsOk = createAction(FETCH_VERSIONS_OK);
+
+
+export function fetch(addonSlug) {
+  /*
+    Fetch add-on.
+  */
+  return (dispatch, getState) => {
+    const apiArgs = getState().apiArgs || {};
+    const addonUrl = Url(
+      urlJoin(process.env.MKT_API_ROOT, 'extensions/extension/', addonSlug)
+    ).q(apiArgs);
+
+    if (process.env.NODE_ENV === 'test') {
```

Comment:
I'm also not sure about wrapping request because it doesn't have anything to do with testing network requests, we just want data.

And yeah, everything's already async here so it doesn't make a difference whether we dispatch within a request callback or not.

I'll go ahead with this *for now*, but I'm totally open to being convinced on something else.